### PR TITLE
T15392

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T15392

Includes content from T15075

### CODE REVIEW

**Comment by cosimoc** [comment]
@smspillaz Can you rebase this branch to the latest master?

**Comment by smspillaz** [comment]
@cosimoc done

File: src/containers.js [hunk 1]
```diff
@@ -313,7 +271,7 @@ const ChatStackChild = new Lang.Class({
                                                   '',
                                                   GObject.ParamFlags.READWRITE |
                                                   GObject.ParamFlags.CONSTRUCT_ONLY,
-                                                  TriggerableEventQueue.$gtype),
+                                                  Queue.TriggerableEventQueue.$gtype),
```
Comment (erikos) [inline src/containers.js]: Should we rename then TriggerableEventQueue to TriggerableEvent as it is already under the Queue namespace?

File: src/containers.js [hunk 2]
```diff
@@ -0,0 +1,443 @@
+// src/containers.js
+//
+// Copyright (c) 2016-2017 Endless Mobile Inc.
+//
+// Containers for chat bubbles.
+//
+// Due to the different interactions that can occurr, we separate out chat
+// bubbles into pairs containers and views, which actually display the content.
```
Comment (erikos) [inline src/containers.js]: This reads odd, "pairs of container and..." maybe?

File: src/containers.js (hunk 1 above)
Comment (smspillaz) [inline src/containers.js]: Probably not - we want to specify that it is a queue of some sort. The duplication in the name is not ideal, but `Queue.TriggerableEvent` doesn't suggests that it is an event emitter and not a queue.

File: src/containers.js (hunk 2 above)
Comment (smspillaz) [inline src/containers.js]: I think I meant to write *paired* containers and views

**Pull request merged**